model-optimizer/extensions/ops/gather.py

```diff
@@ -43,26 +61,31 @@ def infer(node: Node):
         assert data_shape is not None
         indices_shape = node.in_port(1).data.get_shape()
         assert indices_shape is not None
-        axis = node.in_port(2).data.get_value()
-        assert axis is not None
-        axis = get_canonical_axis_index(data_shape, axis)
+        assert node.in_port(2).data.get_value() is not None
+        axis = Gather.get_axis(node)
 
         # we import PermuteInputs locally because it uses Gather inside and we have recursive imports
         from mo.graph.perm_inputs import PermuteInputs
         PermuteInputs().set_input_permutation(node.in_node(1), node, 'input:0', 'axis')
 
+        batch_dims = Gather.get_batch_dims(node)
+        batch_dims_range = indices_shape[:batch_dims]
+        out_shape = np.concatenate((data_shape[:axis], indices_shape[batch_dims:], data_shape[axis + 1:]))
+
         data_value = node.in_port(0).data.get_value()
         indices_value = node.in_port(1).data.get_value()
         if data_value is not None and indices_value is not None:
-            node.out_port(0).data.set_value(np.array(np.take(data_value, int64_array(indices_value), axis),
-                                                     dtype=data_value.dtype))
+            if not batch_dims:
```

Comment:
nit: change to a more explicit version `if batch_dims == 0:`

Reply:
done